### PR TITLE
Updates for Trilinos 14

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes https://github.com/xsdk-project/xsdk-issues/issues/175.

### CODE REVIEW

File: packages/Meshfree/src/DTK_DetailsSplineProlongationOperator.hpp [hunk 1]
```diff
@@ -97,8 +97,10 @@ class SplineProlongationOperator
         using DeviceType = typename Node::device_type;
         using ExecutionSpace = typename DeviceType::execution_space;
 
-        auto x_view = X.getLocalViewDevice();
-        auto y_view = Y.getLocalViewDevice();
+        Y.scale( beta );
+
+        auto x_view = X.getLocalViewDevice(Tpetra::Access::ReadOnly);
+        auto y_view = Y.getLocalViewDevice(Tpetra::Access::ReadWrite);
```
Comment (masterleinad) [inline packages/Meshfree/src/DTK_DetailsSplineProlongationOperator.hpp:103]:
```suggestion
        auto x_view = X.getLocalViewDevice(Tpetra::Access::ReadOnly);
        auto y_view = Y.getLocalViewDevice(Tpetra::Access::ReadWrite);
```

**Comment by masterleinad** [comment]
Retest this please.

File: CMakeLists.txt [hunk 2]
```diff
@@ -64,7 +64,8 @@ ELSE()
         FIND_PATH(_Kokkos_SETTINGS
             NAMES KokkosConfig.cmake
             PATHS ${KOKKOS_PREFIX}
-            ${TPL_Trilinos_LIBRARY_DIRS}/cmake/Kokkos)
+                  ${TPL_Trilinos_LIBRARY_DIRS}/cmake/Kokkos
+                  ${Trilinos_DIR}/../Kokkos)
```
Comment (dalg24) [inline CMakeLists.txt:68]: What is this 2nd path for?

**Review by dalg24 — COMMENTED** [review]
Should we support only the latest Trilinos?
(don't care too much, just trying to understand your rational)

**Comment by masterleinad** [comment]
> Should we support only the latest Trilinos?

Yes, I think it's sufficient to support the latest releases and I'm pretty sure this pull request is not backward-compatible.

**Comment by dalg24** [comment]
> > Should we support only the latest Trilinos?
> 
> Yes, I think it's sufficient to support the latest releases and I'm pretty sure this pull request is not backward-compatible.

Then why guard for the version of Trilinos?

File: CMakeLists.txt (hunk 2 above)
Comment (masterleinad) [reply CMakeLists.txt]: Good question. This was from the original pull request. Let's see if the CI is happy without it.

**Comment by masterleinad** [comment]
> Then why guard for the version of Trilinos?

Ah, also from the original pull request. Fixed now.

**Comment by masterleinad** [comment]
Retest this please.

**Comment by masterleinad** [comment]
Retest this please.

File: CMakeLists.txt (hunk 2 above)
Comment (masterleinad) [reply CMakeLists.txt:68]: We still need it in the CI.

**Review by aprokop — COMMENTED** [review]
LGTM. But a question: if we require just the new version of Trilinos, do we actually check that anywhere in CMake? I don't see any

**Comment by masterleinad** [comment]
> LGTM. But a question: if we require just the new version of Trilinos, do we actually check that anywhere in CMake? I don't see any

I would just specify it in `spack`. I don't know if we ever checked the `Trilinos` version (or `ArborX` version) at configure time.

**Comment by masterleinad** [comment]
I would appreciate it if we can get this merged so that we are ready for the `xSDK` release.

**Review by Rombur — APPROVED** [review]

**Pull request merged**